### ISSUE TITLE
Replace now-deleted VUID-VkRectLayerKHR-offset-01261 with new check

### ISSUE THREAD

**Issue by ianelliottus** [KhronosGroup/Vulkan-ValidationLayers#2757]
**Describe the Issue**
Vulkan spec MR [4442](https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/4442) recently removed VUID-VkRectLayerKHR-offset-01261, and changed the valid usage language per spec issue [2571](https://gitlab.khronos.org/vulkan/vulkan/-/issues/2571).

The `VK_KHR_incremental_present` extension has been modified such that the rectangles are not pre-rotated per `VkSwapchainCreateInfoKHR::preTransform`.  In essence, this means that for a 90/270-degree rotation, the axis of the rectangles in `VkPresentRegionKHR` will be swapped relative to the swapchain's extents.  Thus, the code should change per the following pseudocode:

```
if (swap axis) {
    // Check rectangle x + width against swapchain height
    // Check rectangle y + height against swapchain width
} else {
    // Keep the current code, which does the following:
    // Check rectangle x + width against swapchain width
    // Check rectangle y + height against swapchain height
}
```

**Valid Usage ID**
VUID-VkRectLayerKHR-offset-01261

**Environment:**
 - OS: All, but driven by Android (the only OS to support rotation)
 - GPU: All
 - SDK or header version if building from repo: N/A
 - Options enabled (synchronization, best practices, etc.): N/A

**Comment by mark-lunarg** [comment]
@ianelliottus, the layers were most recently updated to Vulkan-Headers v1.2.174, and the change below arrived in 1.2.175.  I believe an update is imminent and this will be handled as  part of that update.  I'll confirm and add an ETA shortly.


```
-          "vuid": "VUID-VkRectLayerKHR-offset-01261",
-          "text": " The sum of <code>offset</code> and <code>extent</code> <strong class=\"purple\">must</strong> be no greater than the <code>imageExtent</code> member of the <a href=\"#VkSwapchainCreateInfoKHR\">VkSwapchainCreateInfoKHR</a> structure passed to <a href=\"#vkCreateSwapchainKHR\">vkCreateSwapchainKHR</a>"
+          "vuid": "VUID-VkRectLayerKHR-offset-04864",
+          "text": " The sum of <code>offset</code> and <code>extent</code>, after being transformed according to the <code>preTransform</code> member of the <a href=\"#VkSwapchainCreateInfoKHR\">VkSwapchainCreateInfoKHR</a> structure, <strong class=\"purple\">must</strong> be no greater than the <code>imageExtent</code> member of the <a href=\"#VkSwapchainCreateInfoKHR\">VkSwapchainCreateInfoKHR</a> structure passed to <a href=\"#vkCreateSwapchainKHR\">vkCreateSwapchainKHR</a>."
```

**Comment by mark-lunarg** [comment]
I was mistaken, VVL is at 176 and so needs to have this fixed.  Sorry for the misinformation.  The above delta along with Ian's comments above is still pertinent.

**Comment by mark-lunarg** [comment]
Should be addressed by #2758.
